File: dailyprogrammer/2015/11/02/easy.hs
threes :: Int -> String
threes 1 = "1\n"
threes n = (show n) ++ (if n `mod` 3 == 0 then (" 0\n" ++ threes(n `div` 3)) else (if n `mod` 3 == 1 then (" -1\n" ++ threes((n-1) `div` 3)) else (" +1\n" ++ (threes((n+1) `div` 3)))))

main = putStr $ threes 31337357

File: dailyprogrammer/2015/11/02/easy.py
def threes(n):
	if n == 1:
		print n
	else:
		m = n % 3
		x = 0
		if m == 1:
			x = -1
		elif m == 2:
			x = +1

		print n, x
		threes((n + x)/3)

threes(31337357)